src/AlgebraicGeometry/Schemes/ProjectiveSchemes/Objects/Attributes.jl

```diff
@@ -583,3 +583,44 @@ function relative_cotangent_module(X::AbsProjectiveScheme{<:Ring, <:MPolyQuoRing
   return cokernel(psi)
 end
 
+@doc raw"""
+    genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:nor)
+
+Given a projective curve `X` return the genus of `X`, i.e. the 
+integer `p_g = p_a - delta` where `p_a` is the arithmetic genus 
+and `delta` the delta-invariant of the curve.
+
+The `algorithm` keyword can be specified to 
+  - `:normalization` to go via the computation of a normalization
+  - `:primary_decomposition` to proceed with a primary decomposition
+Normalization is usually faster, but not always.
+"""
+function genus(X::AbsProjectiveScheme{<:Field}; algorithm::Symbol=:normalization)
+  @assert dim(X) == 1 "scheme must be one-dimensional"
```

Comment:
```suggestion
  @req dim(X) == 1 "scheme must be one-dimensional"
```
Throwing an `ArgumentError` seems appropriate here